src/main/java/pl/arturzgodka/jsonmappers/SkillMapper.java

```diff
@@ -0,0 +1,70 @@
+package pl.arturzgodka.jsonmappers;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import pl.arturzgodka.apihandlers.BaseUrlParts;
+import pl.arturzgodka.apihandlers.SkillHandlerApi;
+import pl.arturzgodka.datamodel.HeroSkillDataModel;
+import pl.arturzgodka.datamodel.SkillDataModel;
+import pl.arturzgodka.token.FetchToken;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SkillMapper {
+
+    private HeroSkillDataModel fetchSkill(JsonNode node) {
+
+        return new HeroSkillDataModel(
+                node.get("skill").get("name").asText(),
+                node.get("skill").get("level").asInt(),
+                node.get("skill").get("description").asText(),
+                BaseUrlParts.getBaseMediaIconApi() + node.get("skill").get("icon").asText() + ".png",
+                fetchSkillRunes(node)
+        );
+    }
+
+    private HeroSkillDataModel mapSkillToDataModel(String accountData) {
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        JsonNode node = null;
+
+        try {
+            node = objectMapper.readTree(accountData);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+
+        return fetchSkill(node);
+    }
+
+    private List<SkillDataModel> fetchSkillRunes(JsonNode node) {
+
+        List<SkillDataModel> runes = new ArrayList<>();
+
+            for(int i = 0; i < node.get("runes").size(); i++) {
```

Comment:
what if there are no "runes" field in the node?

Reply:
Probably NullPointerExcepiot because trying to get value form api which is null.

Reply:
I added checking "if" "runes" field from node is null.